server/utils/Result.js

```diff
@@ -8,11 +12,13 @@ const Success = value => ({
   },
 
   map(f) {
-    return Success(f(value))
+    const result = f(value)
+    // @ts-ignore
```

Comment:
this is the only `ts-ignore` I couldn't avoid - Success is of type `Success<S,E>`, we need to return `Success<T,E>` and I couldn't work out how to express that